cloud-init/link-nessus-agent.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+"""Link Nessus Agent to Tenable/Nessus server.
+
+This file is a template.  It must be processed by Terraform.
+"""
+
+# Standard Python Libraries
+# Bandit triggers B404 here, but we're only using subprocess.run() and
+# doing so safely.  For more details on B404 see here:
+# https://bandit.readthedocs.io/en/latest/blacklists/blacklist_imports.html#b404-import-subprocess
+import subprocess  # nosec
+import sys
+from typing import Any
+
+# Third-Party Libraries
+import boto3
+
+# Inputs from Terraform
+NESSUS_AGENT_INSTALL_PATH: str = "${nessus_agent_install_path}"
+NESSUS_GROUPS: str = "${nessus_groups}"
+NESSUS_HOSTNAME_KEY: str = "${nessus_hostname_key}"
+NESSUS_KEY_KEY: str = "${nessus_key_key}"
+NESSUS_PORT_KEY: str = "${nessus_port_key}"
+SSM_READ_ROLE_ARN: str = "${ssm_read_role_arn}"
+SSM_REGION: str = "${ssm_region}"
+
+
+def get_parameter(ssm_client, parameter_name: str, with_decryption: bool = True) -> str:
+    """Get the value of the specified parameter."""
+    response: dict[str, Any] = ssm_client.get_parameter(
```

Comment:
Just as a quick mention, using the generics directly for typehinting is technically a Python 3.9+ feature per [PEP-585](https://www.python.org/dev/peps/pep-0585/#backwards-compatibility). Previous versions of Python need to use the the generics in the [typing](https://docs.python.org/3/library/typing.html) library. I don't think it will really impact what's going on in this script, just pointing it out.

Reply:
I can change it if you think it's necessary.  I didn't catch on to that since I haven't done much (if any) type linting before.

Reply:
If we're likely to be on `3.7`+ we could just slap `from __future__ import annotations` at the top to pull in compatible behavior to play it safe.

Reply:
See commit da7bd5a.